src/hip_peer.cpp

```diff
@@ -152,6 +152,8 @@ hipError_t ihipEnablePeerAccess(hipCtx_t peerCtx, unsigned int flags) {
 //---
 hipError_t hipMemcpyPeer(void* dst, hipCtx_t dstCtx, const void* src, hipCtx_t srcCtx,
                          size_t sizeBytes) {
+    if(dstCtx == srcCtx) return hipErrorInvalidValue;
```

Comment:
This check should be after the HIP_INIT_API to ensure HIP tracing & lazy init works correctly. Also return should be wrapped in ihipLogStatus() so that logging works correctly.